### PR TITLE
Unsubdivide and bucket fixes.

### LINKED ISSUE #2: Children don't collapse itself into parent
Even though, all the objects in those 4 children, been removed, still all 4 children don't get removed and collapses itself into parent.

Dump:
```
objects: [[MeshInstance:1218], [MeshInstance:1221]]
child: [Spatial:1227]
  objects: []
  child: Null
  child: Null
  child: Null
  child: Null
child: [Spatial:1228]
  objects: [[MeshInstance:1224], [MeshInstance:1231]]
  child: [Spatial:1237]
    objects: [[MeshInstance:1241], [MeshInstance:1260]]
    child: [Spatial:1269]
      objects: [[MeshInstance:1266], [MeshInstance:1283]]
      child: [Spatial:1314]
        objects: [[MeshInstance:1311]]
        child: Null
        child: Null
        child: Null
        child: Null
      child: [Spatial:1315]
        objects: []
        child: Null
        child: Null
        child: Null
        child: Null
      child: [Spatial:1316]
        objects: []
        child: Null
        child: Null
        child: Null
        child: Null
      child: [Spatial:1317]
        objects: []
        child: Null
        child: Null
        child: Null
        child: Null
    child: [Spatial:1270]
      objects: []
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1271]
      objects: []
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1272]
      objects: [[MeshInstance:1273], [MeshInstance:1308]]
      child: Null
      child: Null
      child: Null
      child: Null
  child: [Spatial:1238]
    objects: [[MeshInstance:1244]]
    child: [Spatial:1279]
      objects: [[MeshInstance:1276]]
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1280]
      objects: []
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1281]
      objects: [[MeshInstance:1302]]
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1282]
      objects: []
      child: Null
      child: Null
      child: Null
      child: Null
  child: [Spatial:1239]
    objects: [[MeshInstance:1234], [MeshInstance:1247]]
    child: [Spatial:1256]
      objects: []
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1257]
      objects: []
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1258]
      objects: [[MeshInstance:1289], [MeshInstance:1305]]
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1259]
      objects: [[MeshInstance:1253]]
      child: Null
      child: Null
      child: Null
      child: Null
  child: [Spatial:1240]
    objects: [[MeshInstance:1263], [MeshInstance:1286]]
    child: [Spatial:1295]
      objects: []
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1296]
      objects: []
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1297]
      objects: []
      child: Null
      child: Null
      child: Null
      child: Null
    child: [Spatial:1298]
      objects: [[MeshInstance:1292], [MeshInstance:1299]]
      child: [Spatial:1321]
        objects: []
        child: Null
        child: Null
        child: Null
        child: Null
      child: [Spatial:1322]
        objects: []
        child: Null
        child: Null
        child: Null
        child: Null
      child: [Spatial:1323]
        objects: []
        child: Null
        child: Null
        child: Null
        child: Null
      child: [Spatial:1324]
        objects: [[MeshInstance:1318]]
        child: Null
        child: Null
        child: Null
        child: Null
child: [Spatial:1229]
  objects: []
  child: Null
  child: Null
  child: Null
  child: Null
child: [Spatial:1230]
  objects: []
  child: Null
  child: Null
  child: Null
  child: Null
                                                                                                            
```

### PR DESCRIPTION
**Pull request by jaynabonne**

### CODE REVIEW

File: addons/sairam.quadtree/QuadTree.gd [hunk 1]
```diff
@@ -195,14 +216,13 @@ func _get_child(body_bounds: AABB) -> QuadTree:
 	
 	Gets the child that incorporates itself in the body_bounds passed.
 	"""
-	if body_bounds.end.z < _bounds.end.z:
-		if body_bounds.end.x < _bounds.end.x:
+	if body_bounds.end.z < _bounds.position.z + _center.z:
```
Comment (sairam4123) [inline addons/sairam.quadtree/QuadTree.gd]: Shouldn't it be `end`?

File: addons/sairam.quadtree/QuadTree.gd [hunk 2]
```diff
@@ -129,7 +140,7 @@ func clear() -> void:
 	if !_is_leaf:  # if the self is not leaf
 		for child in _children:
 			child.clear()  # clear all it's children
-		_is_leaf = true
+		_set_as_empty_leaf()
```
Comment (sairam4123) [inline addons/sairam.quadtree/QuadTree.gd:143]: Why was this moved to a function?

File: addons/sairam.quadtree/QuadTree.gd [hunk 3]
```diff
@@ -48,35 +53,42 @@ func add_body(body: VisualInstance) -> Object:
 		# subdivide if necessary
 		_subdivide()
 		# update the body's quadtree.
-		return update_body(body)
+#		return update_body(body, false)
```
Comment (sairam4123) [inline addons/sairam.quadtree/QuadTree.gd]: There was an issue so I just disabled this line.

File: addons/sairam.quadtree/QuadTree.gd (hunk 1 above)
Comment (jaynabonne) [reply addons/sairam.quadtree/QuadTree.gd]: You want to go up/down and left/right based on splitting the bounds in half. If it's less than half, go down/up, etc.

File: addons/sairam.quadtree/QuadTree.gd (hunk 2 above)
Comment (jaynabonne) [reply addons/sairam.quadtree/QuadTree.gd:143]: It needed to be done in three places, and what was done wasn't trivial - it needed to both set is_leaf *as well as* set all children to null. It seemed to be enough of a concept to warrant its own function. It's more than just setting is_leaf now. That was part of the unsubdivide problem - _children wasn't being set back to all null.

File: addons/sairam.quadtree/QuadTree.gd (hunk 3 above)
Comment (jaynabonne) [reply addons/sairam.quadtree/QuadTree.gd]: Let me look. Without that, it would hang for me.

Comment (sairam4123) [reply addons/sairam.quadtree/QuadTree.gd]: Well, don't forget to update your code.

**Review by sairam4123 — COMMENTED** [review]
Looks fine for me, except Comments.
Please update the comments and add docstrings.

**Review by sairam4123 — APPROVED** [review]
